app/bot/__init__.py

```diff
@@ -11,7 +11,35 @@
 _intents.message_content = True
 
 client = discord.Client(intents=_intents)
+conversation_history = []
 
+# FIXME: This will probably have race condition issues, GL HF :)
+def update_conversation_history(user_message, bot_response):
+    # Append the new user message and bot response to the conversation history
+    conversation_history.append({"role": "user", "content": user_message})
+    conversation_history.append({"role": "bot", "content": bot_response})
+
+    # Optionally, limit the history length to the last N messages
+    if len(conversation_history) > 20:
```

Comment:
We probably should do this. The context is already limited to a certain number of tokens, so there is no reason to store messages that won't be a part of the context

Reply:
It does do this, although I was thinking about how it doesn't have a good way to auto-drop depending on token count. It has no way to know if it's 20 messages that are 2 sentences long, or full on essays. That would be a nice addition (truncate anything beyond token limit automatically).